PetAdoptions/cdk/pet_stack/resources/destroy_stack.sh

```diff
@@ -4,11 +4,168 @@
+            echo "Use -h or --help for usage information"
+            exit 1
+            ;;
+    esac
+done
+
 if [ -z "$AWS_REGION" ]; then
 	echo "Fatal: environment variable AWS_REGION not set. Aborting."
 	exit 1
 fi
 
+# Function to check if stack deletion failed
+check_stack_deletion_status() {
+    local stack_name=$1
+    local status=$(aws cloudformation describe-stacks --stack-name "$stack_name" --query 'Stacks[0].StackStatus' --output text 2>/dev/null)
+
+    if [[ "$status" == "DELETE_FAILED" ]]; then
+        echo "⚠️  Stack $stack_name deletion failed. Status: $status"
+        return 1
+    elif [[ "$status" == "DELETE_COMPLETE" ]] || [[ -z "$status" ]]; then
+        echo "✅ Stack $stack_name deleted successfully or doesn't exist"
+        return 0
+    else
+        echo "ℹ️  Stack $stack_name status: $status"
+        return 0
+    fi
+}
+
+# Function to force delete EKS cluster
+force_delete_eks_cluster() {
+    echo "🔍 Checking for EKS clusters tagged with Workshop=true..."
+
+    local clusters=$(aws eks list-clusters --query 'clusters[]' --output text)
+
+    for cluster in $clusters; do
+        local tags=$(aws eks list-tags-for-resource --resource-arn "arn:aws:eks:$AWS_REGION:$(aws sts get-caller-identity --query Account --output text):cluster/$cluster" --query 'tags.Workshop' --output text 2>/dev/null)
+
+        if [[ "$tags" == "true" ]]; then
+            echo "🗑️  Force deleting EKS cluster: $cluster"
+
+            # Delete node groups first
+            local nodegroups=$(aws eks list-nodegroups --cluster-name "$cluster" --query 'nodegroups[]' --output text 2>/dev/null)
+            for ng in $nodegroups; do
+                echo "   Deleting node group: $ng"
+                aws eks delete-nodegroup --cluster-name "$cluster" --nodegroup-name "$ng" 2>/dev/null
+            done
+
+            # Wait for node groups to be deleted
+            for ng in $nodegroups; do
+                echo "   Waiting for node group $ng to be deleted..."
+                aws eks wait nodegroup-deleted --cluster-name "$cluster" --nodegroup-name "$ng" 2>/dev/null
+            done
+
+            # Delete the cluster
+            aws eks delete-cluster --name "$cluster"
+            echo "   Waiting for cluster $cluster to be deleted..."
+            aws eks wait cluster-deleted --name "$cluster"
+            echo "✅ EKS cluster $cluster deleted"
+        fi
+    done
+}
+
+# Function to force delete VPC and related resources
+force_delete_vpc_resources() {
+    echo "🔍 Checking for VPCs tagged with Workshop=true..."
+
+    local vpcs=$(aws ec2 describe-vpcs --filters "Name=tag:Workshop,Values=true" --query 'Vpcs[].VpcId' --output text)
+
+    for vpc in $vpcs; do
+        echo "🗑️  Force deleting VPC and related resources: $vpc"
+
+        # Delete NAT Gateways
+        local nat_gateways=$(aws ec2 describe-nat-gateways --filter "Name=vpc-id,Values=$vpc" --query 'NatGateways[?State==`available`].NatGatewayId' --output text)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Expressions don&#39;t expand in single quotes, use double quotes for that.